### ISSUE TITLE
Naming of callback typealiases

### ISSUE THREAD

**Issue by teamdisc** [hasancanakgunduz/surveys#16]
It could be just me, but at first, I thought `SurveyListResult` typealias stood for something like

```swift
typealias SurveyListResult = Result<[Survey], Error>
```

But it's a closure, So I think maybe we could add some suffix to make it explicit that it's a closure. Could be something like `SurveyListResultCallback`, or `ResultCompletion<[Survey]>`. What do you think? 🤔

**Comment by hasancanakgunduz** [comment]
I agree, I have also changed the naming from ServiceResult and TokenResult to ServiceResultCallback and TokenResultCallback, thanks ☺️
